paymentsheet/src/main/java/com/stripe/android/paymentsheet/viewmodels/BaseSheetViewModel.kt

```diff
@@ -513,8 +513,14 @@ internal abstract class BaseSheetViewModel(
             updateSelection(null)
         }
 
+        // TODO(samer-stripe): Improve this by either throwing an unexpected error or returning a failure result
+        val requiredConfig = requireNotNull(customerConfig)
```

Comment:
Should we return a Result.failure instead of throwing here?

Reply:
Yeah I think a failure result is best. We should also send an `unexpected_error` evnet here as well. This error should never occur unless we make an accidental logic change that allows it.